src/hal/src/window.rs

```diff
@@ -218,7 +221,20 @@ impl SwapchainConfig {
     /// ```no_run
     ///
     /// ```
-    pub fn with_color<Cf: format::RenderFormat>(mut self) -> Self {
+    pub fn with_color(mut self, cf: format::Format) -> Self {
+        self.color_format = cf;
+        self
+    }
+
+
+    /// Specify the color format for the backbuffer images.
+    ///
+    /// # Examples
+    ///
+    /// ```no_run
+    ///
+    /// ```
+    pub fn with_color_typed<Cf: format::RenderFormat>(mut self) -> Self {
```

Comment:
nice, looks like we are establishing a convention
Now I wonder if mapping API should follow (instead of `_raw`)